internal/cmd/base/servers.go

```diff
@@ -616,18 +618,22 @@ func (b *Server) SetupKMSes(ctx context.Context, ui cli.Ui, config *config.Confi
 			kms.Purpose = origPurpose
 			switch purpose {
 			case globals.KmsPurposeRoot:
-				b.RootKms = wrapper
+				b.RootKms, err = mergeWrappers(ctx, b.RootKms, wrapper)
 			case globals.KmsPurposeWorkerAuth:
-				b.WorkerAuthKms = wrapper
+				b.WorkerAuthKms, err = mergeWrappers(ctx, b.WorkerAuthKms, wrapper)
 			case globals.KmsPurposeWorkerAuthStorage:
-				b.WorkerAuthStorageKms = wrapper
+				b.WorkerAuthStorageKms, err = mergeWrappers(ctx, b.WorkerAuthStorageKms, wrapper)
 			case globals.KmsPurposeRecovery:
-				b.RecoveryKms = wrapper
+				b.RecoveryKms, err = mergeWrappers(ctx, b.RecoveryKms, wrapper)
 			case globals.KmsPurposeConfig:
```

Comment:
if I am understanding this correctly, then we are looping through a config file and creating and setting root kmses based on purpose. the big issue I see here is that with `mergeWrappers` we have an assumption that newer keys will be later in the config. is there anything enforcing this? is it possible that a user could add a new wrapper earlier in the config and accidentally set the _old_ wrapper as the new encrypting key?

Reply:
discussed offline, this is on for mvp if we want to get it into `0.11.x`